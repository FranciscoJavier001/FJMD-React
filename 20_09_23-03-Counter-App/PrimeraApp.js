//** Los componentes se escriben con el UpperCamelCase */
//** Hay dos tipos de componentes, los que estan basados en clases y los que estan basados en funciones (estas se llaman Functional Component) */

import React from 'react';
import PropTypes from 'prop-types'; //** Las Props son las propiedades que el padre manda al hijo */
// import React, { Fragment } from 'react';

const PrimeraApp = ({saludo, subtitulo}) => { //** Functional Components es una funcion */

    return ( //** Como se renderiza a traves de babel es necesario poner el retorno de esta manera para que me regrese algo y despues el fragment (llamado high order component) */
        <> 
        <h1>{saludo}!!!</h1>
            {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */} {/***Si queremos imprimir un arreglo de un objeto esta es la manera de hacerlo */}
            <p>{ subtitulo }</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired //** Asi forzamos a recibir el props de PrimeraApp */
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo' //** AQui directamente lo usamos desde aqui */
}

export default PrimeraApp; //** Aqui se exporta para poderla utilizar */