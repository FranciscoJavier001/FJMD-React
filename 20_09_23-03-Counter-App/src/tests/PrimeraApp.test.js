import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';

import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp />', () => {
    
    // test('debe de mostrar el mensaje "Hola, Soy Francisco ', () => {
    //     const saludo = 'Hola, Soy Francisco';
    //     const {getByText} = render(<PrimeraApp saludo={saludo}/>);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })
    
    test('debe de mostrar <PrimeraApp />> correctamente', () => {

        const saludo = 'Hola, Soy Francisco'; 
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>); //** Es como el Render, para simular clicks y demas cosas como si fuera el querySelector */

       expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar el subtitulo enviado por props ', () => {
        const saludo = 'Hola, Soy Francisco';
        const subTitulo = 'Soy un subtitulo';

        const wrapper = shallow(
        <PrimeraApp 
        saludo={saludo}
        subtitulo={subTitulo}
        />
        );

        const textoParrafo = wrapper.find('p').text(); //** Wrapper tiene toda la informacion del componente renderizado, lo barre con el .find (se parece al querysEelector), luego le dice a donde apunte, por ejemplo, aqui apunta al parrafo "p", luego queremos ver la indo del texto */
        
        expect(textoParrafo).toBe(subTitulo); //** Aqui decimos que textoParrafo tiene que ser un subtitulo */
    });
    
})
