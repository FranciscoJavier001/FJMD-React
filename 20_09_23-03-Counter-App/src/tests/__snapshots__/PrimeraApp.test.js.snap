// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar <PrimeraApp />> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Francisco
    !!!
  </h1>
   
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
